pennylane_forest/qpu.py

```diff
@@ -83,6 +86,28 @@ def __init__(self, device, *, shots=1024, active_reset=True, load_qc=True, reado
 
         self._eigs = {}
 
+        self._compiled_program = None
+        """Union[None, pyquil.ExecutableDesignator]: the latest compiled program. If parametric
+        compilation is turned on, this will be a parametric program."""
+
+        self.parametric_compilation = kwargs.get("parametric_compilation", True)
+
+        if self.parametric_compilation:
+            self._compiled_program_dict = {}
+            """dict[int, pyquil.ExecutableDesignator]: stores circuit hashes associated
+                with the corresponding compiled programs."""
```

Comment:
Since the key is the hash, shouldn't it be a `str`, not an `int`?

Reply:
The hash is an ``int`` that was created by calling on ``hash`` with the serialized ``circuit_graph`` as the argument

Reply:
Ah, of course!